### ISSUE TITLE
Breaking EOFException in bib2gls v2.9 with TeXLive, goes away by downgrading to v2.7

### ISSUE THREAD

**Issue by solarchemist** [nlct/bib2gls#19]
I think I have found abnormal behaviour in bib2gls v2.9 (I should say I have only tested v2.7 and v2.9).

My (not overly complex) document compiles without errors using TeXLive2020 (which has bib2gls v2.7). But after upgrading to TeXLive2022 (which has bib2gls v2.9), the very same document (I made sure no changes occurred, more details below) fails to compile and halts with a bib2gls error (output with `--debug` flag set):

```
Error: EOFException
java.io.EOFException
 at com.dickimawbooks.texparserlib.TeXParser.popStack(TeXParser.java:2197)
 at com.dickimawbooks.texparserlib.TeXObjectList.popArg(TeXObjectList.java:1196)
 at com.dickimawbooks.texparserlib.TeXObjectList.popArg(TeXObjectList.java:1185)
 at com.dickimawbooks.bib2gls.Bib2GlsEntry.initSeeRef(Bib2GlsEntry.java:4318)
 at com.dickimawbooks.bib2gls.Bib2GlsEntry.initCrossRefs(Bib2GlsEntry.java:4306)
 at com.dickimawbooks.bib2gls.GlsResource.processBibList(GlsResource.java:5666)
 at com.dickimawbooks.bib2gls.Bib2Gls.process(Bib2Gls.java:2361)
 at com.dickimawbooks.bib2gls.Bib2Gls.main(Bib2Gls.java:5730)
```

I have been able to reproduce the above error on both TeXLive2021 and TeXLive2022 (bib2gls v2.9 in both cases).
And I can reliably avoid this error by replacing the bib2gls scripts directory of v2.9 with v2.7, for any of TL2020, TL2021 or TL2022. Again, without making any changes to the document.

My stack is TeXLive2022, glossaries-extra 2021/11/22 v1.48, texparserlib.jar 0.9.2.5b (2020-07-11), Java 11.0.15, Ubuntu 18.04.6.

Since my document continues to compile fine even on TL2022 simply by downgrading bib2gls, I suspect something in release 2.8 or 2.9 of bib2gls is causing the error. I'm of course happy to provide more details, including complete source files if necessary, but for now, I'm just looking for a second opinion. Is my analysis of these observations completely off the mark? Or could something be afoot?

Some more details about my document and the glossaries that I hope are pertinent:

+ two glossaries, one for acronyms, the other for symbols
+ LuaLaTeX with [customised class](https://git.solarchemist.se/config/texmf-latex) (based on book)
+ using biber backend for biblatex (for the references)
+ using a large number of LaTeX packages, but I don't suspect any package incompatibility in this case

For completeness, and hoping it might help pinpoint this errant behaviour, I include what I hope is pertinent excerpts from my LaTeX source below. First, my `glossaries-extra` package setup and print commands:
```
\usepackage[%
   record,% record implies undefaction=warn
   abbreviations,%
   symbols,%
   stylemods={longbooktabs},
   nomain]{glossaries-extra}
\GlsXtrLoadResources[%
   src={glossaries/acronyms},%
   selection=all,% because we are in draft mode
   type=abbreviations]
\GlsXtrLoadResources[%
   src={glossaries/symbols},% 
   field-aliases={%
      Value=user1,%  treat Value as user1
      Unit=user2,%   treat Unit as user2      
      Exact=user3,%  this field is used only for exact constants (speed of light, etc.)
      % because using \glssymbol instead of \glsname feels more natural
      % note that this *replaces* \glsname
      % this alias also to avoid having to change existing usage in this document
      Name=symbol},% 
   % symbol sort should be defined *after* field-aliases
   symbol-sort-fallback={symbol},% sort by name field (alias symbol)
   % selection=all because doc is still in draft
   selection=all,% includes all entries, whether used in the doc or not
   type=symbols]
%%%
\printunsrtglossary[%
   type=symbols,%
   style=symbolswithunits,%
   title={List of symbols}]
\printunsrtglossary[%
   type=abbreviations,%
   style=list,%style=list is the default
   title={List of abbreviations}]
```

And here are a few illustrating examples  from my `glossaries/acronyms.bib` and `glossaries/symbols.bib` files:

```
@entry{CB,
  name = {CB},
  description = {conduction band},
  text = {CB},
  long = {conduction band},
  first = {conduction band (CB)}
}
@entry{EBT,
  name = {EBT},
  description = {%
    eriochrome black T is an azo dye,
    $\glssymbol{molar_mass}{=}\SI{461.38}{\gram\per\mol}$,
    systematic name \iupac{sodium 4-[2-(1-hydroxy|naphthalen-2-yl)hydrazin-1-ylidene]-7-nitro-3-oxo-3,4-di|hydro|naphthalene-1-sulfonate}},
  text = {EBT},
  long = {eriochrome black T},
  first = {eriochrome black T (EBT)}
}
```

```
@Symbol{absorption_coefficient,
  Name = {{}\ensuremath{\alpha}},
  See = {},
  Description = {absorption coefficient},
  Text = {absorption coefficient},
  Exact = {},
  Value = {},
  Unit = {},
  First = {absorption coefficient, \ensuremath{\alpha}}
}
@Symbol{elementary_charge,
  Name = {{}\ensuremath{q_\mathrm{e}}},
  See = {},
  Description = {elementary charge},
  Text = {elementary charge},
  Exact = {1},
  Value = {\num{1.602176634e-19}},
  Unit = {\si{\coulomb}},
  First = {elementary charge, \ensuremath{q_\mathrm{e}}}
}
```

I have also defined custom glossary styles, but I'll not include these here for the moment as they are quite verbose, and I have no original work in them anyway, mostly following guides others have published.

In conclusion, let me reiterate that **whatever causes the above bib2gls error goes away by downgrading to v2.7**. I have a fairly complex source document, but I feel confident that I have pinpointed the problem to bib2gls. I'm sorry I haven't provided an MWE - my excuse is that it has already taken me too long to figure out how to work around this problem and putting together this report, furthermore the fact that the behaviour goes away by downgrading bib2gls, and the fact that I have been able to reliably reproduce it many times over different TeXLive releases suggests that the solution is independent of the source file. Still, I could be wrong (I'm inexperienced with Java, for one), and if necessary I can try putting together an MWE.

LaTeX output (excerpts from the bib2gls relevant part) with **bib2gls v2.9** with `--debug` flag (**compilation fails**):
```
bib2gls version 2.9 (2021-11-22)
Reading file:/usr/local/texlive/2022/texmf-dist/scripts/bib2gls/resources/bib2gls-en.xml
Running kpsewhich '--var-value=openin_any'
Processed returned: a 
Running kpsewhich '--var-value=openout_any' 
Processed returned: p 
Running kpsewhich '--var-value=TEXMFOUTPUT' 
Processed returned: . 
Checking write access for thesis.glg 
openin_any=a 
openout_any=p
TEXMFOUTPUT=.
cwd=/media/bay/taha/chepec/thesis 
Checking read access for thesis.log 
6 recognised packages: 
fontenc 
amsmath 
textcomp 
upgreek 
siunitx 
textcase
Checking read access for thesis.aux 
Reading thesis.aux 
Detected font encoding: ts1 
Detected font encoding: pd1 
Detected font encoding: pu 
Adding field: name (name) 
Adding field: sort (sortvalue) 
Adding field: type (type) 
Adding field: first (first) 
Adding field: firstplural (firstpl) 
Adding field: text (text) 
Adding field: plural (plural) 
Adding field: description (desc)
Adding field: descriptionplural (descplural) 
Adding field: symbol (symbol) 
Adding field: symbolplural (symbolplural) 
Adding field: user1 (useri) 
Adding field: user2 (userii) 
Adding field: user3 (useriii) 
Adding field: user4 (useriv) 
Adding field: user5 (userv) 
Adding field: user6 (uservi) 
Adding field: long (long) 
Adding field: longplural (longpl) 
Adding field: short (short) 
Adding field: shortplural (shortpl) 
Adding field: counter (counter) 
Adding field: parent (parent) 
Adding field: loclist (loclist) 
Adding field: location (location) 
Adding field: group (group) 
Adding field: see (see) 
Adding field: alias (alias) 
Adding field: seealso (seealso) 
Adding field: category (category) 
TeX character encoding: UTF-8 
Checking write access for /media/bay/taha/chepec/thesis/thesis.glstex
Initialising resource thesis.glstex. 
Selection mode: all 
Sort mode: en 
Sort field: sort 
Dual sort mode: combine 
Dual sort field: sort 
Label prefix: 
Dual label prefix: dual. 
Tertiary label prefix: tertiary. 
Dual entry mappings: 
descriptionplural -> plural 
plural -> descriptionplural 
name -> description 
description -> name 
Dual symbol mappings: 
symbol -> name 
plural -> symbolplural 
symbolplural -> plural 
name -> symbol 
Dual abbreviation mappings: 
duallong -> long 
longplural -> duallongplural 
dualshort -> short 
short -> dualshort 
dualshortplural -> shortplural 
shortplural -> dualshortplural 
long -> duallong 
duallongplural -> longplural 
Dual abbreviation-entry mappings: 
longplural -> plural 
short -> text 
long -> name 
Dual index-entry mappings: 
name -> name 
Dual index-symbol mappings: 
symbol -> name 
plural -> symbolplural 
symbolplural -> plural 
name -> symbol 
Dual index-abbreviation mappings: 
name -> name 
Checking write access for /media/bay/taha/chepec/thesis/thesis-1.glstex 
Initialising resource thesis-1.glstex. 
Selection mode: all 
Sort mode: en 
Sort field: sort 
Dual sort mode: combine 
Dual sort field: sort 
Label prefix: 
Dual label prefix: dual. 
Tertiary label prefix: tertiary. 
Dual entry mappings: 
descriptionplural -> plural 
plural -> descriptionplural 
name -> description 
description -> name 
Dual symbol mappings: 
symbol -> name 
plural -> symbolplural 
symbolplural -> plural 
name -> symbol 
Dual abbreviation mappings: 
duallong -> long 
longplural -> duallongplural 
dualshort -> short 
short -> dualshort 
dualshortplural -> shortplural 
shortplural -> dualshortplural 
long -> duallong 
duallongplural -> longplural 
Dual abbreviation-entry mappings: 
longplural -> plural 
short -> text 
long -> name 
Dual index-entry mappings: 
name -> name 
Dual index-symbol mappings: 
symbol -> name 
plural -> symbolplural
symbolplural -> plural 
name -> symbol 
Dual index-abbreviation mappings: 
name -> name 
Parsing bib files for resource thesis.glstex. 
Checking read access for glossaries/acronyms.bib 
Reading glossaries/acronyms.bib 
Encoding: UTF-8 
Parsing bib files for resource thesis-1.glstex. 
Checking read access for glossaries/symbols.bib 
Reading glossaries/symbols.bib 
Processing resource thesis.glstex. 
>> name={CBE} 
=>> name={CBE} 
>> first={conduction band edge (CBE)} 
=>> first={conduction band edge (CBE)} 
>> text={CBE} 
=>> text={CBE} 
>> description={conduction band edge} 
=>> description={conduction band edge} 
>> long={conduction band edge} 
=>> long={conduction band edge} 
>> see={CB} 
=>> see={CB}
>> name={EBT} 
=>> name={EBT} 
>> first={eriochrome black T (EBT)} 
=>> first={eriochrome black T (EBT)} 
>> text={EBT} 
=>> text={EBT} 
>> description={% 
 eriochrome black T is an azo dye,
 $\glssymbol{molar_mass}{=}\SI{461.38}{\gram\per\mol}$,
 systematic name \iupac{sodium 4-[2-(1-hydroxy|naphthalen-2-yl)hydrazin-1-ylidene]-7-nitro-3-oxo-3,4-di|hydro|naphthalene-1-sulfonate}} 
=>> description={% 
 eriochrome black T is an azo dye,
 $\glssymbol{molar_mass}{=}\SI{461.38}{\gram\per\mol}$,
 systematic name \iupac{sodium 4-[2-(1-hydroxy|naphthalen-2-yl)hydrazin-1-ylidene]-7-nitro-3-oxo-3,4-di|hydro|naphthalene-1-sulfonate}} 
>> long={eriochrome black T} 
=>> long={eriochrome black T} 
Adding record {EBT}{}{page}{glsnumberformat}{63} to entry EBT's record list. 
[...]
Processing resource thesis-1.glstex.
>> first={absorption coefficient, \ensuremath{\alpha}}
=>> first={absorption coefficient, \ensuremath{\alpha}}
>> text={absorption coefficient}
=>> text={absorption coefficient}
>> description={absorption coefficient}
=>> description={absorption coefficient}
>> symbol={{}\ensuremath{\alpha}}
=>> symbol={{}\ensuremath{\alpha}}
>> user1={}
=>> user1={}
>> user2={}
=>> user2={}
>> user3={}
=>> user3={}
>> see={}
=>> see={}
Warning: Entry 'absorption_coefficient' missing required field: name
Error: EOFException
java.io.EOFException
 at com.dickimawbooks.texparserlib.TeXParser.popStack(TeXParser.java:2197)
 at com.dickimawbooks.texparserlib.TeXObjectList.popArg(TeXObjectList.java:1196)
 at com.dickimawbooks.texparserlib.TeXObjectList.popArg(TeXObjectList.java:1185)
 at com.dickimawbooks.bib2gls.Bib2GlsEntry.initSeeRef(Bib2GlsEntry.java:4318)
 at com.dickimawbooks.bib2gls.Bib2GlsEntry.initCrossRefs(Bib2GlsEntry.java:4306)
 at com.dickimawbooks.bib2gls.GlsResource.processBibList(GlsResource.java:5666)
 at com.dickimawbooks.bib2gls.Bib2Gls.process(Bib2Gls.java:2361)
 at com.dickimawbooks.bib2gls.Bib2Gls.main(Bib2Gls.java:5730)
Transcript written to thesis.glg.
Latexmk: In running custom-dependency rule
 to make 'thesis.glstex' from 'thesis.aux'
 function 'run_bib2gls' did not make the destination.
```

The compilation appears to choke on the very first entry in the symbol glossary (this is exactly the same `EOFException` as shown up top).

LaTeX output (excerpts from the bib2gls relevant part) with **bib2gls v2.7** with `--debug` flag (**compilation succeeds**):
```
bib2gls version 2.7 (2020-07-11)                                                                                                  
Reading file:/usr/local/texlive/2022/texmf-dist/scripts/bib2gls/resources/bib2gls-en.xml
Running kpsewhich '--var-value=openin_any'
Processed returned: a
Running kpsewhich '--var-value=openout_any'
Processed returned: p
Running kpsewhich '--var-value=TEXMFOUTPUT'
Processed returned: .
Checking write access for thesis.glg
openin_any=a
openout_any=p
TEXMFOUTPUT=.
cwd=/media/bay/taha/chepec/thesis
Checking read access for thesis.log
6 recognised packages:
fontenc
amsmath
textcomp
upgreek
siunitx
textcase
Checking read access for thesis.aux                                                                                               
Reading thesis.aux
Detected font encoding: ts1                                                                                                       
Detected font encoding: pd1                                                                                                       
Detected font encoding: pu                                                                                                        
Adding field: name (name)                                                                                                         
Adding field: sort (sortvalue)                                                                                                    
Adding field: type (type)                                                                                                         
Adding field: first (first)                                                                                                       
Adding field: firstplural (firstpl)                                                                                               
Adding field: text (text)                                                                                                         
Adding field: plural (plural)                                                                                                     
Adding field: description (desc)                                                                                                  
Adding field: descriptionplural (descplural)                                                                                      
Adding field: symbol (symbol)                                                                                                     
Adding field: symbolplural (symbolplural)                                                                                         
Adding field: user1 (useri)                                                                                                       
Adding field: user2 (userii)                                                                                                      
Adding field: user3 (useriii)                                                                                                     
Adding field: user4 (useriv)                                                                                                      
Adding field: user5 (userv)                                                                                                       
Adding field: user6 (uservi)                                                                                                      
Adding field: long (long)                                                                                                         
Adding field: longplural (longpl)                                                                                                 
Adding field: short (short)                                                                                                       
Adding field: shortplural (shortpl)                                                                                               
Adding field: counter (counter)                                                                                                   
Adding field: parent (parent)                                                                                                     
Adding field: loclist (loclist)                                                                                                   
Adding field: location (location)                                                                                                 
Adding field: group (group)                                                                                                       
Adding field: see (see)                                                                                                           
Adding field: alias (alias)                                                                                                       
Adding field: seealso (seealso)                                                                                                   
Adding field: category (category)                                                                                                 
TeX character encoding: UTF-8                                                                                                     
Checking write access for /media/bay/taha/chepec/thesis/thesis.glstex                                                             
Initialising resource thesis.glstex.                                                                                              
Selection mode: all
Sort mode: en
Sort field: sort
Dual sort mode: combine
Dual sort field: sort
Label prefix:
Dual label prefix: dual.
Tertiary label prefix: tertiary.
Dual entry mappings:
descriptionplural -> plural
plural -> descriptionplural
name -> description
description -> name
Dual symbol mappings:
symbol -> name
plural -> symbolplural
symbolplural -> plural
name -> symbol
Dual abbreviation mappings:
duallong -> long
longplural -> duallongplural
dualshort -> short
short -> dualshort
dualshortplural -> shortplural
shortplural -> dualshortplural
long -> duallong
duallongplural -> longplural
Dual abbreviation-entry mappings:
longplural -> plural
short -> text
long -> name
Dual index-entry mappings:
name -> name
Dual index-symbol mappings:
symbol -> name
plural -> symbolplural
symbolplural -> plural
name -> symbol
Dual index-abbreviation mappings:
name -> name
Checking write access for /media/bay/taha/chepec/thesis/thesis-1.glstex
Initialising resource thesis-1.glstex.
Selection mode: all
Sort mode: en
Sort field: sort
Dual sort mode: combine
Dual sort field: sort
Label prefix:
Dual label prefix: dual.
Tertiary label prefix: tertiary.
Dual entry mappings:
descriptionplural -> plural
plural -> descriptionplural
name -> description
description -> name
Dual symbol mappings:
symbol -> name
plural -> symbolplural
symbolplural -> plural
name -> symbol
Dual abbreviation mappings:
duallong -> long
longplural -> duallongplural
dualshort -> short
short -> dualshort
dualshortplural -> shortplural
shortplural -> dualshortplural
long -> duallong
duallongplural -> longplural
Dual abbreviation-entry mappings:
longplural -> plural
short -> text
long -> name
Dual index-entry mappings:
name -> name
Dual index-symbol mappings:
symbol -> name
plural -> symbolplural
symbolplural -> plural
name -> symbol
Dual index-abbreviation mappings:
name -> name
Parsing bib files for resource thesis.glstex.
Checking read access for glossaries/acronyms.bib
Reading glossaries/acronyms.bib
Encoding: UTF-8
Parsing bib files for resource thesis-1.glstex.
Checking read access for glossaries/symbols.bib
Reading glossaries/symbols.bib
Processing resource thesis.glstex.
>> name={CBE}                                                                                                                     
=>> name={CBE}                                                                                                                    
>> first={conduction band edge (CBE)}                                                                                             
=>> first={conduction band edge (CBE)}                                                                                            
>> text={CBE}                                                                                                                     
=>> text={CBE}                                                                                                                    
>> description={conduction band edge}                                                                                             
=>> description={conduction band edge}                                                                                            
>> long={conduction band edge}                                                                                                    
=>> long={conduction band edge}                                                                                                   
>> see={CB}                                                                                                                       
=>> see={CB}                                                                                                                      
>> name={EBT}                                                                                                                     
=>> name={EBT}                                                                                                                    
>> first={eriochrome black T (EBT)}
=>> first={eriochrome black T (EBT)}
>> text={EBT}                                                                                                                     
=>> text={EBT}                                                                                                                    
>> description={%                                                                                                                 
    eriochrome black T is an azo dye,
    $\glssymbol{molar_mass}{=}\SI{461.38}{\gram\per\mol}$,
    systematic name \iupac{sodium 4-[2-(1-hydroxy|naphthalen-2-yl)hydrazin-1-ylidene]-7-nitro-3-oxo-3,4-di|hydro|naphthalene-1-sulfonate}}
=>> description={%
    eriochrome black T is an azo dye,
    $\glssymbol{molar_mass}{=}\SI{461.38}{\gram\per\mol}$,
    systematic name \iupac{sodium 4-[2-(1-hydroxy|naphthalen-2-yl)hydrazin-1-ylidene]-7-nitro-3-oxo-3,4-di|hydro|naphthalene-1-sulfonate}}
>> long={eriochrome black T}                                                                                                      
=>> long={eriochrome black T}                                                                                                     
Adding record {EBT}{}{page}{glsnumberformat}{67} to entry EBT's record list.                                                      
[...]
Processing resource thesis-1.glstex.
>> first={absorption coefficient, \ensuremath{\alpha}}
=>> first={absorption coefficient, \ensuremath{\alpha}}
>> text={absorption coefficient}
=>> text={absorption coefficient}
>> description={absorption coefficient}
=>> description={absorption coefficient}
>> symbol={{}\ensuremath{\alpha}}
=>> symbol={{}\ensuremath{\alpha}}
>> user1={}
=>> user1={}
>> user2={}
=>> user2={}
>> user3={}
=>> user3={}
>> see={}
=>> see={}
Warning: Entry 'absorption_coefficient' missing required field: name
>> first={elementary charge, \ensuremath{q_\mathrm{e}}}                                                                           
=>> first={elementary charge, \ensuremath{q_\mathrm{e}}}                                                                          
>> text={elementary charge}                                                                                                       
=>> text={elementary charge}                                                                                                      
>> description={elementary charge}                                                                                                
=>> description={elementary charge}                                                                                               
>> symbol={{}\ensuremath{q_\mathrm{e}}}                                                                                           
=>> symbol={{}\ensuremath{q_\mathrm{e}}}                                                                                          
>> user1={\num{1.602176634e-19}}                                                                                                  
=>> user1={\num{1.602176634e-19}}                                                                                                 
>> user2={\si{\coulomb}}                                                                                                          
=>> user2={\si{\coulomb}}                                                                                                         
>> user3={1}                                                                                                                      
=>> user3={1}                                                                                                                     
>> see={}                                                                                                                         
=>> see={}                                                                                                                        
Warning: Entry 'elementary_charge' missing required field: name                                                                   
[...]
texparserlib:--> 𝛼                                                                                                                
Break points: 𝛼|.                                                                                                                 
absorption_coefficient -> '𝛼|' (𝛼) [-128 0 -40 54 -34 -3 0 66 0 0 0 0]                                                            
Parsing code block:
{}\ensuremath{T}
texparserlib:--> T                                                                                                                
Break points: T|.                                                                                                                 
absolute_temperature -> 'T|' (T) [0 103 0 66 0 0 0 0]                                                                             
Parsing code block:
{}\ensuremath{E_\mathrm{g}}
texparserlib:                                                                                                                     
EOFException                                                                                                                      
java.io.EOFException                                                                                                              
        at com.dickimawbooks.texparserlib.TeXParser.popNextArg(TeXParser.java:1942)                                               
        at com.dickimawbooks.texparserlib.TeXParser.popNextArg(TeXParser.java:1927)                                               
        at com.dickimawbooks.texparserlib.latex.MathFontCommand.process(MathFontCommand.java:41)                                  
        at com.dickimawbooks.texparserlib.TeXCsRef.process(TeXCsRef.java:103)                                                     
        at com.dickimawbooks.texparserlib.html.L2HConverter.subscript(L2HConverter.java:986)
        at com.dickimawbooks.texparserlib.SbChar.process(SbChar.java:69)
        at com.dickimawbooks.texparserlib.html.L2HMathGroup.processList(L2HMathGroup.java:72)
        at com.dickimawbooks.texparserlib.Group.process(Group.java:85)
        at com.dickimawbooks.texparserlib.latex.EnsureMath.process(EnsureMath.java:65)
        at com.dickimawbooks.texparserlib.TeXCsRef.process(TeXCsRef.java:103)
        at com.dickimawbooks.bib2gls.Bib2Gls.interpret(Bib2Gls.java:1341)
        at com.dickimawbooks.bib2gls.SortComparator.updateSortValue(SortComparator.java:421)
        at com.dickimawbooks.bib2gls.Bib2GlsEntryComparator.updateSortValue(Bib2GlsEntryComparator.java:163)
        at com.dickimawbooks.bib2gls.SortComparator.sortEntries(SortComparator.java:756)
        at com.dickimawbooks.bib2gls.Bib2GlsEntryComparator.sortEntries(Bib2GlsEntryComparator.java:444)
        at com.dickimawbooks.bib2gls.GlsResource.sortData(GlsResource.java:6389)
        at com.dickimawbooks.bib2gls.GlsResource.sortData(GlsResource.java:6324)
        at com.dickimawbooks.bib2gls.GlsResource.sortDataIfRequired(GlsResource.java:6297)
        at com.dickimawbooks.bib2gls.GlsResource.processBibData(GlsResource.java:6528)
        at com.dickimawbooks.bib2gls.GlsResource.processData(GlsResource.java:5478)
        at com.dickimawbooks.bib2gls.Bib2Gls.process(Bib2Gls.java:2227)
        at com.dickimawbooks.bib2gls.Bib2Gls.main(Bib2Gls.java:5249)                                                              
Break points: ensuremath|E|mathrm|g|.                                                                                             
Eg -> 'ensuremath|E|mathrm|g|' (E) [0 88 0 97 0 102 0 104 0 101 0 88 0 96 0 83 0 103 0 91 0 66 0 88 0 66 0 96 0 83 0 103 0 91 0 101 0 96 0 66 0 90 0 66 0 0 0 0]
```

The debug output from the succesful compilation contained lots of these `EOFException` messages from `texparserlib`, but  the document compiles OK and the glossary in the document behaves normally (linked with hyperref, etc.).

Those warnings about `missing required field: name` for each and every entry in the symbol glossary was perplexing, and I did some testing to better understand it, but for now assume it's not related to this abnormal behaviour (if it turns out to be, I'll report my findings then).

Phew! That was a long one :-) Thanks for reading, and hoping the problem is more readily apparent to you than it was to me.

**Comment by nlct** [comment]
I think the problem may just stem from the `name` field being required for the given entry type. You're aliasing `Name` to `symbol` so there's no actual `name` field set. I suggest copying another field into it, even if you don't need the actual field in your document. For example, you could use `replicate-fields={name=symbol}` so the symbol is in both `name` and `symbol`.

I suspect the reason for the change in behaviour between the two versions results from the bug fix in v2.8:

>  `@entry` missing required name/description results in "null"
    saved in the field value. This has been changed to an empty
    string (but the warning about the missing field remains).

I expect that if you look at the `.glstex` file for the v2.7 version you'll find the `name` field set to `null` for `absorption_coefficient`. Hopefully, simply assigning `name` should fix the problem, but the missing field warning shouldn't cause the process to fail.

The `EOFException` is always thrown when the end of the input stream is reached, but it's normally caught and shouldn't cause `bib2gls` to fail. I've been making extensive changes to the TeX Parser library (which is where the exception is coming from) and there is also a new version of `bib2gls` pending (which is waiting for the updates to the TeX parser library and a new version of `glossaries-extra`). I'll try some missing field tests to see if I can reproduce the problem.

**Comment by nlct** [comment]
This should now be fixed in v3.0. Any required fields that are missing should now just generate a warning from `bib2gls` and the field will be set to empty in the `.glstex` file.
